compiler/package.go

```diff
@@ -127,6 +137,35 @@ type funcContext struct {
 	pos token.Pos
 }
 
+// newRootCtx creates a new package-level instance of a functionContext object.
+func newRootCtx(srcs sources, typesInfo *types.Info, typesPkg *types.Package, isBlocking func(*types.Func) bool, minify bool) *funcContext {
+	pkgInfo := analysis.AnalyzePkg(srcs.Files, srcs.FileSet, typesInfo, typesPkg, isBlocking)
+	rootCtx := &funcContext{
+		parent:   nil, // Package-level context has no parent.
+		FuncInfo: pkgInfo.InitFuncInfo,
+		pkgCtx: &pkgContext{
+			Info:                 pkgInfo,
+			additionalSelections: make(map[*ast.SelectorExpr]selection),
+
+			pkgVars:      make(map[string]string),
+			objectNames:  make(map[types.Object]string),
+			varPtrNames:  make(map[*types.Var]string),
+			escapingVars: make(map[*types.Var]bool),
+			indentation:  1,
+			minify:       minify,
+			fileSet:      srcs.FileSet,
+		},
+		allVars:     make(map[string]int),
+		flowDatas:   map[*types.Label]*flowData{nil: {}},
+		caseCounter: 1,
+		labelCases:  make(map[*types.Label]int),
+	}
+	for name := range reservedKeywords {
+		rootCtx.allVars[name] = 1
```

Comment:
What is the significance of `1` here?

Reply:
It's documented in the allVars godoc: the number of visible Go objects with that exact name potentially accessible from the current scope (e.g. due to variable shadowing). By setting this value to `1` we won't assign the reserved keyword to aby JS object if we encounter a synonymous Go object, but rather `keyword$1`.